sdk/keyvault/azure-security-keyvault/azure/security/keyvault/keys/_client.py

```diff
@@ -3,266 +3,715 @@
+        permission.
 
-        key = self._deserialize("Key", response)
-
-        return key
-
-    def update_key(self, name, version, key_ops=None, attributes=None, tags=None, **kwargs):
-        # type: (str, str, Optional[List[str]], Mapping[str, str], Mapping[str, str], Any) -> Key
-        url = "/".join([self.vault_url, "keys", name, version])
-
-        headers = {"Content-Type": "application/json; charset=utf-8", "x-ms-client-request-id": str(uuid.uuid1())}
-
-        update_params = KeyUpdateParameters(key_ops=key_ops, key_attributes=attributes, tags=tags)
-        body = self._serialize.body(update_params, "KeyUpdateParameters")
-        request = HttpRequest("PATCH", url, headers=headers)
-        request.set_json_body(body)
-        request.format_parameters({"api-version": self.API_VERSION})
-        response = self._pipeline.run(request, **kwargs).http_response
-        if response.status_code != 200:
-            raise HttpResponseError("Request failed with code {}: '{}'".format(response.status_code, response.text()))
-        key = self._deserialize("Key", response)
-
-        return key
-
-    def _internal_paging(self, url, max_page_size, next_link=None, raw=False, **kwargs):
-        # type: (str, int, Optional[str], Optional[bool], Any) -> HttpResponse
-        if next_link:
-            url = next_link
-            query_parameters = {}
+        :param name: The name of the deleted key.
+        :type name: str
+        :returns: The recovered deleted key
+        :rtype: ~azure.security.keyvault.keys._models.Key
+
+        Example:
+            .. literalinclude:: ../tests/test_examples_keys.py
+                :start-after: [START recover_deleted_key]
+                :end-before: [END recover_deleted_key]
+                :language: python
+                :dedent: 4
+                :caption: Recovers the specified soft-deleted key
+        """
+        bundle = self._client.recover_deleted_key(self.vault_url, name)
+        return Key._from_key_bundle(bundle)
+
+    def update_key(self, name, version, key_ops=None, enabled=None, expires=None, not_before=None, tags=None, **kwargs):
+        # type: (str, str, Optional[List[str]], Optional[bool], Optional[datetime], Optional[datetime], Optional[Dict[str, str]], Mapping[str, Any]) -> Key
+        """The update key operation changes specified attributes of a stored key
+        and can be applied to any key type and key version stored in Azure Key
+        Vault.
+
+        In order to perform this operation, the key must already exist in the
+        Key Vault. Note: The cryptographic material of a key itself cannot be
+        changed. This operation requires the keys/update permission.
+
+        :param name: The name of key to update.
+        :type name
+        :param version: The version of the key to update.
+        :type version
+        :param key_ops: Json web key operations. For more information on
```

Comment:
Consider `key_operations` after API review comments with the name

Reply:
Same as [this](https://github.com/Azure/azure-sdk-for-python/pull/5437/files?file-filters%5B%5D=.py#r288677273)